drivers/pinctrl/pinctrl_silabs_siwx917.c

```diff
@@ -1,25 +1,48 @@
 /*
  * Copyright (c) 2023 Antmicro
+ * Copyright (c) 2024 Silicon Laboratories Inc.
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #define DT_DRV_COMPAT silabs_siwx917_pinctrl
 
 #include <zephyr/drivers/pinctrl.h>
-#include "rsi_rom_egpio.h"
+#include "sl_si91x_peripheral_gpio.h"
 
-static void pinctrl_siwx917_set(bool use_ulp, uint8_t port, uint8_t pin, uint8_t mux)
+#define MODE_COUNT               16
+#define HP_PERIPHERAL_ON_ULP_PIN 6
+
+static inline bool valid_mode(uint8_t mode)
```

Comment:
The compiler will probably inline this function anyway, especially when optimizing for size. But I like specifying it anyway, as it communicates intent: This is effectively just a named logic expression, it could have been repeated 3 times at the different call sites, or been a preprocessor macro. It should always be inlined.

Reply:
I'd have had the same comment as Jérôme, i.e. normally the only valid case for using `inline` is if you've actually first proven that the compiler really isn't doing the right thing. However, I understand your point about wanting to highlight the "similar to a utility macro" intent, so I'm fine either way here.

Reply:
Yeah, I shared your opinion in the past. Then, one reviewer taught to me that when a reviewer see a call to `valid_mode()`, he'll think `valid_mode()` is a global function provided by Zephyr.
That said, I have no strong opinion (and in this this case, it is really obvious because the code is really short).